pcm-iio.cpp

```diff
@@ -333,11 +333,71 @@ vector<string> build_display(vector<struct iio_stacks_on_socket>& iios, vector<s
     return buffer;
 }
 
-void display(const vector<string> &buff)
+std::string build_csv_row(const std::vector<std::string>& chunks, const std::string& delimeter) 
+{
+    return std::accumulate(chunks.begin(), chunks.end(), std::string(""), 
+                           [delimeter](const string &left, const string &right){
+                               return left.empty() ? right : left + delimeter + right;
+                           });
+}
+
+vector<string> build_csv(vector<struct iio_stacks_on_socket>& iios, vector<struct counter>& ctrs, 
+    const bool human_readable, const std::string& csv_delimeter)
+{
+    vector<string> result;
+    vector<string> current_row;
+    auto header = combine_stack_name_and_counter_names("Name");
+    header.insert(header.begin(), "Socket");
+    result.push_back(build_csv_row(header, csv_delimeter));
+    std::map<uint32_t,map<uint32_t,struct counter*>> v_sort;
+    //re-organize data collection to be row wise
+    size_t max_name_width = 0;
+    for (std::vector<struct counter>::iterator counter = ctrs.begin(); counter != ctrs.end(); ++counter) {
+        v_sort[counter->v_id][counter->h_id] = &(*counter);
+        max_name_width = std::max(max_name_width, counter->v_event_name.size());
```

Comment:
Windows MSVC compiler has a known issue with std::max . Compilation on appveyor fails:

c:\projects\pcm\pcm-iio_win\../pcm-iio.cpp(357): error C2589: '(': illegal token on right side of '::' [C:\projects\pcm\PCM-IIO_Win\pcm-iio-win.vcxproj]

tr y to fix it by putting std::max in brackets (std::max)(max_name_width, counter->v_event_name.size());